app/models/account_statuses_filter.rb

```diff
@@ -55,7 +55,14 @@ def filtered_scope
   end
 
   def filtered_reblogs_scope
-    Status.left_outer_joins(:reblog).where(reblog_of_id: nil).or(Status.where.not(reblogs_statuses: { account_id: current_account.excluded_from_timeline_account_ids }))
+    Status
+      .left_outer_joins(:reblog).where(reblog_of_id: nil)
+      .joins('LEFT OUTER JOIN accounts reblogs_accounts ON reblogs_accounts.id = reblogs_statuses.account_id')
```

Comment:
I'm a bit confused by the position of the `where` here, wouldn't it be more understandable like this?
```suggestion
      .left_outer_joins(:reblog)
      .joins('LEFT OUTER JOIN accounts reblogs_accounts ON reblogs_accounts.id = reblogs_statuses.account_id')
      .where(reblog_of_id: nil)
```

An even more concise and idiomatic way may be:
```suggestion
    .left_outer_joins(reblog: :account)
    .where(reblog_of_id: nil)
```

Though that would rename `reblogs_accounts` to `accounts` and that may end up more confusing and more brittle.